Tests/TuistKitTests/Dot/MockDotGraphGenerator.swift

```diff
@@ -0,0 +1,24 @@
+import Basic
+import Foundation
+import TuistGenerator
+
+final class MockDotGraphGenerator: DotGraphGenerating {
+    var generateProjectCallCount: UInt = 0
```

Comment:
minor: we can check the array counts to obtain the same information, it helps reduce number of items to manage in the mock, but is mostly stylistic so not an issue. (e.g. `generateProjectArgs.count`)